### PR TITLE
Split out static auth methods from Auth object

### PR DESCRIPTION
**Pull request by erikjohnston**
Part of https://github.com/matrix-org/synapse/pull/1808

### CODE REVIEW

File: synapse/api/auth.py [hunk 1]
```diff
@@ -43,30 +42,9 @@
...
-        auth_events = yield self.store.get_events(auth_events_ids)
-        auth_events = {
-            (e.type, e.state_key): e for e in auth_events.values()
-        }
-        self.check(event, auth_events=auth_events, do_sig_check=do_sig_check)
-
-    def check(self, event, auth_events, do_sig_check=True):
+class Auther(object):
+    @staticmethod
+    def check(event, auth_events, do_sig_check=True):
```
Comment (NegativeMjark) [inline synapse/api/auth.py]: Would it make sense to make these top level functions rather than staticmethods?

Comment (erikjohnston) [inline synapse/api/auth.py]: Maybe? Would you prefer that? I somewhat like having all the helper functions for `check` bundled together

File: synapse/api/auth.py [hunk 2]
```diff
@@ -504,84 +614,197 @@ def _verify_third_party_invite(self, event, auth_events):
...
+            auth_events (dict: event-key -> event): the existing room state.
+
+
+        Returns:
+            True if the auth checks pass.
+        """
+        with Measure(self.clock, "auth.check"):
+            Auther.check(event, auth_events, do_sig_check=do_sig_check)
+
+    def check_size_limits(self, event):
```
Comment (NegativeMjark) [inline synapse/api/auth.py]: Isn't check_size_limits defined as a staticmethod on the Auther class above?

File: synapse/api/auth.py (hunk 1 above)
Comment (NegativeMjark) [inline synapse/api/auth.py]: I would prefer that. I would also prefer it we took this opportunity to move the event auth functions to a separate file.

Comment (NegativeMjark) [inline synapse/api/auth.py]: (which would go someway towards separating the code for authing CS interactions from code for authing events)

Comment (erikjohnston) [inline synapse/api/auth.py]: Done

File: synapse/api/auth.py (hunk 2 above)
Comment (erikjohnston) [inline synapse/api/auth.py]: Done

File: synapse/api/auth.py [hunk 3]
```diff
@@ -975,54 +562,7 @@ def compute_auth_events(self, event, current_state_ids, for_verification=False):
         defer.returnValue(auth_ids)
 
     def _get_send_level(self, etype, state_key, auth_events):
```
Comment (NegativeMjark) [inline synapse/api/auth.py]: Is _get_send_level called from anywhere? I can't see a reference to it.

**Pull request merged**